api/controllers/AuthController.js

```diff
@@ -28,6 +28,27 @@ module.exports = {
     passport.authenticate('reddit', {
       state: req.session.state,
       duration: 'permanent',
+      failureRedirect: '/login',
+      scope: 'identity'
+    })(req, res);
+  },
+
+  modAuth: function(req, res) {
+    req.session.state = crypto.randomBytes(32).toString('hex');
+    if (req.query.url) {
+      req.session.redirectUrl = req.query.url;
+    }
+    passport.authenticate('reddit', {
+      state: req.session.state,
+      duration: 'permanent',
+      failureRedirect: '/login',
+    })(req, res);
```

Comment:
So I figured out what was causing that callback error from before.

It seems like passport-reddit (the oauth-2 authentication strategy) will use the scope that it finds in express.js by default. This can be overridden by specifying a scope in the passport.authenticate() function. 

Separate point: The lowest-level reddit scope is 'identity', which gives the app access to someone's reddit username. Since the 'identity' scope is necessary for any other scope to mean anything, passport-reddit always uses the 'identity' scope along with any other scopes that you provide. ([Source code](https://github.com/Slotos/passport-reddit/blob/master/lib/passport-reddit/strategy.js#L51))

Anyway, as mentioned above, the default scope from express.js can be overridden by specifying a scope in the passport.authenticate() function. However, if you override the default scope with anything other than the string 'identity', it returns a callback error.

 \>_>